File: agent/src/main/java/com.evgenysobko.diploma.agent/AgentMain.java
package com.evgenysobko.diploma.agent;

import java.lang.instrument.Instrumentation;

public class AgentMain {

    public static Instrumentation savedInstrumentationInstance;

    public static void premain(String agentArgs, Instrumentation instrumentation) {
        agentmain(agentArgs, instrumentation);
    }

    public static void agentmain(String agentArgs, Instrumentation instrumentation) {
        savedInstrumentationInstance = instrumentation;
    }
}

File: src/main/kotlin/com/evgenysobko/diploma/agent/AgentMain.kt
package com.evgenysobko.diploma.agent

import java.lang.instrument.Instrumentation

class AgentMain {

    companion object {

        lateinit var savedInstrumentationInstance: Instrumentation

        fun premain(agentArgs: String, instrumentation: Instrumentation) = agentMain(agentArgs, instrumentation)

        fun agentMain(agentArgs: String, instrumentation: Instrumentation) {
            savedInstrumentationInstance = instrumentation
        }
    }
}